backend/server.js

```diff
@@ -118,6 +118,62 @@ app.get("/top-tracks", async (req, res) => {
 });
 
 
+app.get("/compatible-tracks", async (req, res) => {
+  try {
+    // get the most listened track
+    const topTracksResponse = await spotifyApi.getMyTopTracks({ limit: 1, time_range: 'short_term' });
+    const topTrack = topTracksResponse.body.items[0];
+    const trackId = topTrack.id;  
   
-  
+    // get audio features
+    const audioFeaturesResponse = await spotifyApi.getAudioFeaturesForTrack(trackId);
+    const { key, mode, tempo } = audioFeaturesResponse.body;
+
+    // find compatible track with the following audio features
+    const minTempo = tempo - 5;
+    const maxTempo = tempo + 5;
+    const compatibleTracksResponse = await spotifyApi.searchTracks(`key:${key} mode:${mode} tempo:${minTempo}-${maxTempo}`, { limit: 1 });
```

Comment:
this seems to return a not so compatible song in testing, a rock song seems compatible with an spa ambient music? sent you links of the songs on slack. maybe we find a better logic